### PR TITLE
Implementation of the ATM term

### PR DESCRIPTION
**Pull request by marvinfriede**
- [x] ATM single and batched
- [x] tests (including grad)
- [x] better typing (parameters should be tensors)

I also added pre-commit hooks, tests for Python 3.10, and transitioned typing to the 3.10+ style (with `from __future__ import annotations` to support older versions).

### CODE REVIEW

**Review by awvwgk — CHANGES_REQUESTED** [review]
> I also added pre-commit hooks, tests for Python 3.10, and transitioned typing to the 3.10+ style (with `from __future__ import annotations` to support older versions).

After some experimentation with this future import I found that it in fact does not enable support for older Python versions. The annotation parsing is just deferred and will not fail anymore at runtime if invalid. Rather, without running a type checker it is not possible to tell whether the annotations are correct.

Type checking should be supported in all Python versions and not only 3.10+, therefore I would kindly request to revert this change.

File: src/tad_dftd3/damping/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""
+Collection of damping schemes.
+"""
+from .atm import *
+from .rational import *
```
Comment (awvwgk) [inline src/tad_dftd3/damping/__init__.py:5]: Can we make these import explicit?

Comment (marvinfriede) [reply src/tad_dftd3/damping/__init__.py:5]: Why? The corresponding modules have `__all__` to avoid possible pollution by importing with `*`.

Comment (awvwgk) [reply src/tad_dftd3/damping/__init__.py:5]: The usage of ``__all__`` is currently inconsistent in the code base, not all modules define it. Also, there is a single function exported from ``.rational``, which does not require the usage of ``__all__``:

https://github.com/dftd3/tad-dftd3/blob/a4580ff0cec31a081ea276c8ebe0774dec5eb1ea/src/tad_dftd3/damping/rational.py#L9

Comment (marvinfriede) [reply src/tad_dftd3/damping/__init__.py]: Changed for consistency.

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. Looks good to me.

**Pull request merged**